src/Compilers/Core/Portable/SourceGeneration/Nodes/PredicateSyntaxStrategy.cs

```diff
@@ -48,7 +48,7 @@ public Builder(PredicateSyntaxStrategy<T> owner, object key, StateTableStore tab
                 _name = name;
                 _comparer = comparer;
                 _key = key;
-                _filterTable = table.GetStateTableOrEmpty<SyntaxNode>(_owner._filterKey).ToBuilder(stepName: null, trackIncrementalSteps);
+                _filterTable = table.GetStateTableOrEmpty<SyntaxNode>(_owner._filterKey).ToBuilder(stepName: null, trackIncrementalSteps, equalityComparer: Roslyn.Utilities.ReferenceEqualityComparer.Instance);
```

Comment:
`T` is not constrainted to `class` so why do we use a reference equality comparer vs. simply the default comparer?

Reply:
This was code written by Cyrus originally, I'm refactoring it to keep it semantically the same.